### DISCUSSION TITLE
Offline mode question

### DISCUSSION THREAD

**Discussion by yuriydrobniy** [rnmapbox/maps#2784] [Q&A]
How can I implement offline mode for my case where I use dynamic `styleJSON` for manipulating of showing layers based on a user preferences?

My `styleJSON` contain a general styles and dynamic sources and layers:
```
import mapboxStyleJson from './mapbox-style.json';

const myStyleJSON = {
  ...mapboxStyleJson,
  sources: {
    ...mapboxStyleJson.sources,
    ...userLayers.sources,
  },
  layers: {
    ...mapboxStyleJson.layers,
    ...userLayers.layers,
  },
};
```
To use offline mode, I need to save the pack using `createPack` and pass the `styleURL` there. In theory, I can separate general styles and other sources and layers. The general styles I can pass like `styleURL` (can get it from studio.mapbox) and make an offline pack from it. But then I don't know how to pass these sources and layers to show dynamic user data. 

thnks in advance.

**Comment by Cdingram** [comment]
Can't figure this out either
